src/Fixer/ConfigurationDefinitionFixerInterface.php

```diff
@@ -15,6 +15,8 @@
 use PhpCsFixer\OptionsResolver;
 
 /**
+ * Maybe better the other way around, i.e. now there is no upgrade path to the new interface without deprecation warnings
```

Comment:
It would not solve the problem: no upgrade path without deprecations, even if the deprecations are only there for a weak or so.